### ISSUE TITLE
Softmax/Sigmoid required?

### ISSUE THREAD

**Issue by rohan1561** [jacobgil/pytorch-grad-cam#232]
I am confused as to whether or not I need to retain the softmax/sigmoid layers at the output for my model or remove them. I see several applications where these were retained but the paper suggests that the outputs need to come "before the softmax". Kindly clarify

**Comment by jacobgil** [comment]
Typically you would not use softmax/sigmoid at all with grad-cam.
The reason is then you are selecting features that maximize the response for a certain category, but you care less about features that also minimize the response of other categories.
Once you add softmax, features that minimize the response from other categories might contribute at all.

Having said that - it's probably worth experimenting with both.

**Comment by jacobgil** [comment]
Closing for now, please reopen if you have more questions.
